### PR TITLE
Add proxy support

### PR DESCRIPTION
**Pull request by hdep**
Hi,

Following #57 
this PR add proxy_address for download_file. The variable name is proxy_url.
If undef we use dowload_file without proxy which is default.

Regards,

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -64,14 +66,16 @@
   $package_name            = $nsclient::params::package_name,
   $download_destination    = $nsclient::params::download_destination,
   $config_template         = $nsclient::params::config_template,
-  $install_path            = $nsclient::params::install_path
+  $install_path            = $nsclient::params::install_path,
+  $proxy_url               = $nsclient::params::proxy_url
```
Comment (bastelfreak) [inline manifests/init.pp]: As in the other PR, please add a datatype and remove the validate_string() function.

File: manifests/install.pp [hunk 2]
```diff
@@ -25,10 +26,23 @@
         }
       }
 
-      download_file { 'NSCP-Installer':
-        url                   => $source,
-        destination_directory => $nsclient::download_destination,
-        require               => File[$nsclient::download_destination],
+      if $nsclient::proxy_url {
```
Comment (bastelfreak) [inline manifests/install.pp]: this if/else block isn't needed. You can always pass the `$nsclient::proxy_url` variable. It won't do anything if it is undef.

**Comment by bastelfreak** [comment]
Hi @hdep, can you please do a fixup/squash to melt all commits into one? If you need some help with git or writing tests, you can always ping us in our IRC channel #voxpupuli on freenode or on slack at https://slack.puppet.com

File: manifests/install.pp (hunk 2 above)
Comment (hdep) [reply manifests/install.pp]: Ok I wasn't sure. Will check this.

File: manifests/init.pp [hunk 3]
```diff
@@ -64,7 +66,8 @@
   $package_name            = $nsclient::params::package_name,
   $download_destination    = $nsclient::params::download_destination,
   $config_template         = $nsclient::params::config_template,
-  $install_path            = $nsclient::params::install_path
+  $install_path            = $nsclient::params::install_path,
+  String $proxy_url        = $nsclient::params::proxy_url
```
Comment (bastelfreak) [inline manifests/init.pp]: This need to be optional:
```suggestion
  Optional[String] $proxy_url = $nsclient::params::proxy_url
```

**Comment by hdep** [comment]
here is the squash.

File: manifests/init.pp [hunk 4]
```diff
@@ -73,6 +76,7 @@
   validate_string($config_template)
   validate_string($install_path)
 
+
```
[the remark this answers is not in the document]
Comment (hdep) [reply manifests/init.pp]: I'll remove it but this line is on master branch.

**Comment by bastelfreak** [comment]
Thanks for the work!

**Pull request merged**